requre/cassette.py

```diff
@@ -21,14 +21,670 @@
+                "Requre can't work in this setup: we are meant to read "
+                "recorded responses but the storage file does not exist."
+            )
+        if self.mode == StorageMode.read:
+            return False
+
+        elif self.mode == StorageMode.write:
+            return True
+
+        elif self.mode == StorageMode.append:
+            if self.data_miner.data_type in [DataTypes.DictWithList, DataTypes.List]:
+                # append possible
+                return True
+
+            if (
+                self.data_miner.data_type in [DataTypes.Dict, DataTypes.Value]
+                and keys not in self
+            ):
+                # keys present => we can read the values
+                return True
+            return False
+
+        key_present = keys in self
+        logger.warning(
+            f"No storage-mode set. Checking presence of the keys: {key_present} "
+            f"=> {'read' if key_present else 'store'}"
+        )
+        return not key_present
+
+    @storage_file_version.setter  # type: ignore
+    def storage_file_version(self, value: int):
+        self.metadata = {self.version_key: value}
+
+    def _set_storage_metadata_if_not_set(self):
+        """
+        Set storage file version if not already set to latest version.
+        """
+        if self.metadata.get(self.version_key) is None:
+            self.storage_file_version = VERSION_REQURE_FILE
+        if self.metadata.get(self.key_inspect_strategy_key) is None:
+            self.metadata = {
+                self.key_inspect_strategy_key: self.data_miner.key_stategy_cls.__name__
+            }
+        data_type_name = DataTypes.__name__
+        if self.metadata.get(data_type_name) is None:
+            self.metadata = {data_type_name: self.data_miner.data_type.value}
+
+    @property
+    def storage_file(self):
+        return self._storage_file
+
+    @storage_file.setter
+    def storage_file(self, value):
+        # when set to None, reset to default
+        if value is None:
+            self._set_defaults()
+            return
+        # when file is changed set PersistenStorage default values
+        if self._storage_file != value:
+            self._set_defaults()
+            self._storage_file = value
+
+        if not os.path.exists(self._storage_file):
+            if self.mode == StorageMode.default:
+                self.mode = StorageMode.write
+
+            elif self.mode == StorageMode.read:
+                raise PersistentStorageException(
+                    "Requre can't work in this setup: we are meant to read"
+                    f" recorded responses but the storage file ({self._storage_file}) "
+                    "does not exist."
+                )
+        else:
+            if self.mode == StorageMode.default:
+                self.mode = StorageMode.read
+            self.storage_object = self.load()
+
+    @staticmethod
+    def transform_hashable(keys: List) -> List:
```

Comment:
Is this relevant to this class? Can we have it as some helper function somewhere else?

Reply:
Still relevant, could you please move this one out of the class or create an issue after mering. (I just don't want to forget about those small issues that make the reviews harder.)